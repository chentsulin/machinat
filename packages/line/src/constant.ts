export const CHANNEL_REQUEST_GETTER = Symbol(
  'channel_request_getter.line.machinat'
);
export const BULK_REQUEST_GETTER = Symbol('bulk_request_getter.line.machinat');

/** @internal */
export const LINE = 'line' as const;

/** @internal */
export const PATH_REPLY = 'v2/bot/message/reply';

/** @internal */
export const PATH_PUSH = 'v2/bot/message/push';

/** @internal */
export const PATH_MULTICAST = 'v2/bot/message/multicast';

/** @internal */
export const PATH_RICHMENU = 'v2/bot/richmenu';

export enum LiffContextOs {
  Ios,
  Android,
  Web,
}
