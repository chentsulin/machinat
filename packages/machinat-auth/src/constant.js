// @flow
import { namedInterface } from '@machinat/core/service';
import type { ServerAuthorizer, AuthModuleConfigs } from './types';

export const STATE_COOKIE_KEY = 'machinat_auth_state';
export const TOKEN_COOKIE_KEY = 'machinat_auth_token';
export const SIGNATURE_COOKIE_KEY = 'machinat_auth_signature';
export const ERROR_COOKIE_KEY = 'machinat_auth_error';

export const AUTH_SERVER_AUTHORIZERS_I = namedInterface<
  ServerAuthorizer<any, any>[]
>({
  name: 'ServerAuthorizersList',
  multi: true,
});

export const AUTH_MODULE_CONFIGS_I = namedInterface<AuthModuleConfigs>({
  name: 'AuthModuleConfigs',
});
