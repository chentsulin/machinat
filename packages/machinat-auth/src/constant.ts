import { makeInterface } from '@machinat/core/service';
import type { ServerAuthorizer, AuthModuleConfigs } from './types';

export const STATE_COOKIE_KEY = 'machinat_auth_state';
export const TOKEN_COOKIE_KEY = 'machinat_auth_token';
export const SIGNATURE_COOKIE_KEY = 'machinat_auth_signature';
export const ERROR_COOKIE_KEY = 'machinat_auth_error';

export const SERVER_AUTHORIZERS_I = makeInterface<ServerAuthorizer<any, any>[]>(
  { name: 'ServerAuthorizersList', multi: true }
);

export const MODULE_CONFIGS_I = makeInterface<AuthModuleConfigs>({
  name: 'AuthModuleConfigs',
});
