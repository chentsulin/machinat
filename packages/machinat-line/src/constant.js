// @flow
export const LINE = 'line';
export const LINE_NAMESPACE = 'Line';

export const LINE_NATIVE_TYPE = Symbol.for('machinat.line.native');
