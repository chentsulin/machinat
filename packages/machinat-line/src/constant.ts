export const LINE = 'line';

export const CHANNEL_API_CALL_GETTER = Symbol(
  'line.segment.channel_api_call_getter'
);
export const BULK_API_CALL_GETTER = Symbol('line.segment.bulk_api_call_getter');

export const PATH_REPLY = 'v2/bot/message/reply';
export const PATH_PUSH = 'v2/bot/message/push';
export const PATH_MULTICAST = 'v2/bot/message/multicast';
export const PATH_RICHMENU = 'v2/bot/richmenu';
