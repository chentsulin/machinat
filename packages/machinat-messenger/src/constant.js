// @flow
export const MESSENGER = 'messenger';
export const MESSENGER_NAMESPACE = 'Messenger';
export const MESSENGER_NATIVE_TYPE = Symbol.for('machinat.messenger.native');

export const ATTACHED_FILE_DATA = Symbol(
  'machinat.messenger.job.attached_file_data'
);

export const ATTACHED_FILE_INFO = Symbol(
  'machinat.messenger.job.attached_file_info'
);

export const ENTRY_MESSAGES = 'me/messages';

export const ENTRY_PASS_THREAD_CONTROL = 'me/pass_thread_control';
export const ENTRY_REQUEST_THREAD_CONTROL = 'me/request_thread_control';
export const ENTRY_TAKE_THREAD_CONTROL = 'me/take_thread_control';

export const ENTRY_MESSAGE_CREATIVES = 'me/message_creatives';
export const ENTRY_BROADCAST_MESSAGES = 'me/broadcast_messages';
export const ENTRY_CUSTOM_LABELS = 'me/custom_labels';
