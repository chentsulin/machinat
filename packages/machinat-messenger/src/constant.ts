export const MESSENGER = 'messenger';

export const ATTACHMENT_DATA = Symbol('messenger.segment.attachment_data');

export const ATTACHMENT_INFO = Symbol('messenger.segment.attachment_info');

export const ASSET_TAG = Symbol.for('messenger.segment.asset_tag');
export const ENTRY_PATH = Symbol.for('messenger.segment.entry');

export const PATH_MESSAGES = 'me/messages';

export const PATH_PASS_THREAD_CONTROL = 'me/pass_thread_control';
export const PATH_REQUEST_THREAD_CONTROL = 'me/request_thread_control';
export const PATH_TAKE_THREAD_CONTROL = 'me/take_thread_control';

export const PATH_MESSAGE_ATTACHMENTS = 'me/message_attachments';
export const PATH_MESSENGER_PROFILE = 'me/messenger_profile';
export const PATH_PERSONAS = 'me/personas';
