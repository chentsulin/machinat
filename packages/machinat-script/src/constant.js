// @flow
export const MACHINAT_SCRIPT_TYPE = Symbol.for('machinat.component.script');

export const SCRIPT_STATE_KEY = '$$machinat:script';
