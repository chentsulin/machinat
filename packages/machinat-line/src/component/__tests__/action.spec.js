import Machinat from 'machinat';
import {
  PostbackAction,
  MessageAction,
  URIAction,
  DateTimePickerAction,
  CameraAction,
  CameraRollAction,
  LocationAction,
} from '../action';
import { LINE_NATIVE_TYPE } from '../../constant';
import renderHelper from './renderHelper';

const render = renderHelper(() => null);

test.each(
  [
    PostbackAction,
    MessageAction,
    URIAction,
    DateTimePickerAction,
    CameraAction,
    CameraAction,
    CameraRollAction,
    LocationAction,
  ].map(C => [C.name, C])
)('%s is valid native Component', (_, Action) => {
  expect(typeof Action).toBe('function');
  expect(Action.$$native).toBe(LINE_NATIVE_TYPE);
  expect(Action.$$getEntry).toBe(undefined);
});

it.each(
  [
    <PostbackAction data="__POSTBACK_FOO__" label="Hello!" text="WORLD!" />,
    <MessageAction label="Tick" text="Tock" />,
    <URIAction uri="http://machinat.com" label="Try it!" />,
    <DateTimePickerAction
      label="Which day?"
      data="__MEETUP_DATE__"
      date
      initial="2018-01-01"
      min={new Date('2000-01-01')}
      max={new Date('2046-01-01')}
    />,
    <DateTimePickerAction
      label="What time?"
      data="__MEETUP_TIME__"
      time
      initial="00:00"
      min={new Date('2000 12:12')}
      max={new Date('2000 21:21')}
    />,
    <DateTimePickerAction
      label="Say it again?"
      data="__MEETUP_DATETIME__"
      initial="2000-01-01T00:00"
      min={new Date('1990-01-01T00:00')}
      max={new Date('2020-01-01T00:00')}
    />,
    <CameraAction label="Cheer!" />,
    <CameraRollAction label="Cheer again!" />,
    <LocationAction label="Ok, where are we?" />,
  ].map(element => [element.type.name, element])
)('%s match snapshot', (_, actionElement) => {
  const segments = render(actionElement);
  expect(segments.length).toBe(1);

  const segment = segments[0];
  expect(segment.type).toBe('part');
  expect(segment.node).toBe(actionElement);
  expect(segment.path).toBe('$');
  expect(segment.value).toMatchSnapshot();
});

describe('DateTimePickerAction', () => {
  test('datetime mode', () => {
    [
      <DateTimePickerAction
        mode="datetime"
        initial="2000-01-01T00:00"
        min={new Date(1990, 0)}
        max={new Date(2020, 0)}
      />,
      <DateTimePickerAction
        initial="2000-01-01T00:00"
        min={new Date(1990, 0)}
        max={new Date(2020, 0)}
      />,
      <DateTimePickerAction
        date
        time
        initial="2000-01-01T00:00"
        min={new Date(1990, 0)}
        max={new Date(2020, 0)}
      />,
    ].forEach(pickerElement => {
      const [{ value }] = render(pickerElement);

      expect(value.mode).toBe('datetime');
      expect(value.initial).toBe('2000-01-01T00:00');
      expect(value.min).toBe('1990-01-01T00:00');
      expect(value.max).toBe('2020-01-01T00:00');
    });
  });

  test('date mode', () => {
    [
      <DateTimePickerAction
        mode="date"
        initial="2000-01-01"
        min={new Date(1990, 0)}
        max={new Date(2020, 0)}
      />,
      <DateTimePickerAction
        date
        initial="2000-01-01"
        min={new Date(1990, 0)}
        max={new Date(2020, 0)}
      />,
    ].forEach(pickerElement => {
      const [{ value }] = render(pickerElement);

      expect(value.mode).toBe('date');
      expect(value.initial).toBe('2000-01-01');
      expect(value.min).toBe('1990-01-01');
      expect(value.max).toBe('2020-01-01');
    });
  });

  test('time mode', () => {
    [
      <DateTimePickerAction
        mode="time"
        initial="11:11"
        min={new Date(0, 0, 0, 0, 0)}
        max={new Date(0, 0, 0, 22, 22)}
      />,
      <DateTimePickerAction
        time
        initial="11:11"
        min={new Date(0, 0, 0, 0, 0)}
        max={new Date(0, 0, 0, 22, 22)}
      />,
    ].forEach(pickerElement => {
      const [{ value }] = render(pickerElement);

      expect(value.mode).toBe('time');
      expect(value.initial).toBe('11:11');
      expect(value.min).toBe('00:00');
      expect(value.max).toBe('22:22');
    });
  });
});
