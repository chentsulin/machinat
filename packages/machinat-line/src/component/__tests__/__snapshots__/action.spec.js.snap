// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CameraAction match snapshot 1`] = `
Object {
  "label": "Cheer!",
  "type": "camera",
}
`;

exports[`CameraRollAction match snapshot 1`] = `
Object {
  "label": "Cheer again!",
  "type": "cameraRoll",
}
`;

exports[`DateTimePickerAction match snapshot 1`] = `
Object {
  "data": "__MEETUP_DATE__",
  "initial": "2018-01-01",
  "label": "Which day?",
  "max": "2046-01-01",
  "min": "2000-01-01",
  "mode": "date",
  "type": "datetimepicker",
}
`;

exports[`DateTimePickerAction match snapshot 2`] = `
Object {
  "data": "__MEETUP_TIME__",
  "initial": "00:00",
  "label": "What time?",
  "max": "21:21",
  "min": "12:12",
  "mode": "time",
  "type": "datetimepicker",
}
`;

exports[`DateTimePickerAction match snapshot 3`] = `
Object {
  "data": "__MEETUP_DATETIME__",
  "initial": "2000-01-01T00:00",
  "label": "Say it again?",
  "max": "2020-01-01T00:00",
  "min": "1990-01-01T00:00",
  "mode": "datetime",
  "type": "datetimepicker",
}
`;

exports[`LocationAction match snapshot 1`] = `
Object {
  "label": "Ok, where are we?",
  "type": "location",
}
`;

exports[`MessageAction match snapshot 1`] = `
Object {
  "label": "Tick",
  "text": "Tock",
  "type": "message",
}
`;

exports[`PostbackAction match snapshot 1`] = `
Object {
  "data": "__POSTBACK_FOO__",
  "displayText": "WORLD!",
  "label": "Hello!",
  "type": "postback",
}
`;

exports[`URIAction match snapshot 1`] = `
Object {
  "label": "Try it!",
  "type": "uri",
  "uri": "http://machinat.com",
}
`;
